test/e2e_node/oomkiller_linux_test.go

```diff
@@ -150,6 +166,61 @@ func runOomKillerTest(f *framework.Framework, testCase testCase, kubeReservedMem
 	})
 }
 
+func runOomKillerFailsToKillTest(f *framework.Framework, testCase testCase, kubeReservedMemory float64) {
+	ginkgo.Context(testCase.name, func() {
+		if kubeReservedMemory > 0 && testCase.enableSingleProcessKill {
+			framework.Failf("only one of kubeReservedMemory and enableSingleProcessKill can be set")
+		}
+
+		// Update KubeReservedMemory in KubeletConfig.
+		if kubeReservedMemory > 0 {
+			tempSetCurrentKubeletConfig(f, func(ctx context.Context, initialConfig *kubeletconfig.KubeletConfiguration) {
+				if initialConfig.KubeReserved == nil {
+					initialConfig.KubeReserved = map[string]string{}
+				}
+				// There's a race condition observed between system OOM and cgroup OOM if node alocatable
+				// memory is equal to node capacity. Hence, reserving a fraction of node's memory capacity for
+				// K8s components such that node allocatable memory is less than node capacity to
+				// observe OOM kills at cgroup level instead of system OOM kills.
+				initialConfig.KubeReserved["memory"] = fmt.Sprintf("%d", int(kubeReservedMemory*getLocalNode(context.TODO(), f).Status.Capacity.Memory().AsApproximateFloat64()))
+			})
+		}
+		if testCase.enableSingleProcessKill {
```

Comment:
if we're only using this function when this is true, do we need to branch here? or: can we consolidate the two functions and branch on this variable?

Reply:
Either is good with me, updated to branch in the test function.